### DISCUSSION TITLE
limiting users to certain locations/departments/etc

### DISCUSSION THREAD

**Discussion by BioMedGPH** [Samedis-care/samedis-care-community#34] [Ideas]
It would be great if I could set things up to limit users to certain locations, so they could see status of devices, status of maintenance, or make problem reports for only their location. Maybe it can be based on their groups (matching device and user group)

**Comment by YvesR** [comment]
Can you explain with a few examples what you have exactly in mind?

**Reply by BioMedGPH** [reply]
One use case is to have the Operating Room in-charge (or one of the Ward in-charge, Lab in-charge, etc) having a Samedis account, And then this person would be allowed to see the devices and repair/maintenance tasks for devices that are assigned to him. Either based on device location(s) (Operating Room 1, Operating room 2, Recovery Area, etc), based on Department(s) that own the device, or based on a certain defined group(s) of devices. So this person can not see all the devices in the hospital, but only devices that are relevant to him/her.  Nurse in-charge could see all devices in her nursing ward. And someone higher up like the head of nursing could maybe see all devices all nursing wards. In my opinion this level control needs to be flexible, based on multiple locations/departments/device groups/user groups.

I think the current access control only controls the level of access, based on function of the person, but cannot control which devices can be seen or not.

**Reply by YvesR** [reply]
Indeed, at the moment we do have roles that can or can not access modules and lists/data inside. Currently there is no row-level-authorization as this is a big matter of performance and logic.
We do need also consider if we do not want to see other devices, data, etc. or may not see it. 
Filters can provide the first option, the second one is more complex as you already mentioned that there are many different logical considerations.
